src/builder/QuarkBuilder.sol

```diff
@@ -312,4 +337,13 @@ contract QuarkBuilder {
         }
         return result;
     }
+
+    function findMaxCost(Payment memory payment, uint256 chainId) internal pure returns (uint256) {
+        for (uint256 i = 0; i < payment.maxCosts.length; ++i) {
+            if (payment.maxCosts[i].chainId == chainId) {
+                return payment.maxCosts[i].amount;
+            }
+        }
+        return DEFAULT_MAX_PAYCALL_COST;
```

Comment:
I think the behavior we want is to actually ignore chains or throw an error if max cost is not provided for a chain.

Reply:
Ok that sounds good 👍